exercises/nodejs/src/game.js

```diff
@@ -1,172 +1,198 @@
-import generator from 'random-seed'
+import generator from "random-seed";
 
-var Game = function () {
-  var players = new Array()
-  var places = new Array(6)
-  var purses = new Array(6)
-  var inPenaltyBox = new Array(6)
+let Game = function () {
+  const players = new Array();
+  const boardPositions = new Array(6);
+  const purses = new Array(6);
+  const inPenaltyBox = new Array(6);
 
-  var popQuestions = new Array()
-  var scienceQuestions = new Array()
-  var sportsQuestions = new Array()
-  var rockQuestions = new Array()
+  const popQuestions = new Array();
+  const scienceQuestions = new Array();
+  const sportsQuestions = new Array();
+  const rockQuestions = new Array();
 
-  var currentPlayer = 0
-  var isGettingOutOfPenaltyBox = false
+  const minimumPlayers = 2;
+  const numberOfCatagories = 4;
 
-  var didPlayerWin = function () {
-    return !(purses[currentPlayer] == 6)
-  }
+  let currentPlayer = 0;
+  let isGettingOutOfPenaltyBox = false;
 
-  var currentCategory = function () {
-    if (places[currentPlayer] == 0) { return 'Pop' }
-    if (places[currentPlayer] == 4) { return 'Pop' }
-    if (places[currentPlayer] == 8) { return 'Pop' }
-    if (places[currentPlayer] == 1) { return 'Science' }
-    if (places[currentPlayer] == 5) { return 'Science' }
-    if (places[currentPlayer] == 9) { return 'Science' }
-    if (places[currentPlayer] == 2) { return 'Sports' }
-    if (places[currentPlayer] == 6) { return 'Sports' }
-    if (places[currentPlayer] == 10) { return 'Sports' }
-    return 'Rock'
-  }
+  let didPlayerWin = function () {
+    return !(purses[currentPlayer] == 6);
+  };
 
-  this.createRockQuestion = function (index) {
-    return 'Rock Question ' + index
-  }
+  let currentCategory = function () {
+    if (boardPositions[currentPlayer] % numberOfCatagories === 0 ) {
+      return "Pop";
+    }
+    if (boardPositions[currentPlayer] % numberOfCatagories === 1) {
+      return "Science";
+    }
+    if (boardPositions[currentPlayer] % numberOfCatagories === 2) {
+      return "Sports";
+    }
+    return "Rock";
+  };
 
-  for (var i = 0; i < 50; i++) {
-    popQuestions.push('Pop Question ' + i)
-    scienceQuestions.push('Science Question ' + i)
-    sportsQuestions.push('Sports Question ' + i)
-    rockQuestions.push(this.createRockQuestion(i))
+  this.createRockQuestion = function (index) {
+    return "Rock Question " + index;
+  };
+
+  for (let i = 0; i < 50; i++) {
+    popQuestions.push("Pop Question " + i);
+    scienceQuestions.push("Science Question " + i);
+    sportsQuestions.push("Sports Question " + i);
+    rockQuestions.push(this.createRockQuestion(i));
```

Comment:
It was hard to see what have been changed here at the beginning because those were mostly things related to code formatting and some of the code was moved down to other lines that was detected as a change. But the codes looks better with semicolons, double quotes and var instead of let.